src/platform/api/pythonApi.ts

```diff
@@ -468,4 +485,32 @@ export class InterpreterService implements IInterpreterService {
             })
             .catch(noop);
     }
+    private getRefreshPromise(): Promise<void> | undefined {
+        if (this.canFindRefreshPromise === undefined) {
+            this.canFindRefreshPromise = false;
+            const api = this.tryGetApi();
+            if (!api) {
+                this.getApi()
```

Comment:
`tryGetApi` will already have called `getApi`, so this code block doesn't feel necessary

Reply:
The reason it needs to be called is to force the update of the 'canFindRefreshPromise' which getApi doesn't do.

Reply:
The then clause.

Reply:
getRefreshPromise technically has to be synchronous. So it can't await because then the function itself would be part of the refresh promise.

This makes the order like so:

1. Somebody calls 'refreshing' getter
2. 'Refreshing' getter calls this function to get a refresh promise or not
3. canFindRefreshPromise is undefined
4. canFindRefreshPromise is set to false
5. API is not initialized, so trygetApi returns undefined
6. canFindRefreshPromise is still false, so undefined is returned from the function
7. A promise is scheduled that updates canFindRefreshPromise
8. Next call to 'refreshing' uses the new value.

The root of all this is the weird 'getRefreshPromise() : Promise<void> | undefined that the python extension has. If they'd just always returned a promise, and made the promise resolve when doing refreshing, this could have been a lot simpler.

Reply:
I see where I'm confused - in step 5, tryGetApi sees that the api is undefined so it calls getApi, but **doesn't wait for it**. 
That will cause 2 calls to `this.apiProvider.getApi()` per call to this function though, right? Is it fine to stack up those calls?